src/app/network/UploadFolderManager.ts

```diff
@@ -165,7 +163,13 @@ export class UploadFoldersManager {
     (task, next: (err: Error | null, res?: DriveFolderData) => void) => {
       if (this.abortController?.signal.aborted) return;
 
-      this.manageMemoryUsage();
+      const newConcurrency = QueueUtilsService.instance.getConcurrencyUsingPerfomance(
+        this.uploadFoldersQueue.concurrency,
+        UploadFoldersManager.MAX_CONCURRENT_UPLOADS,
+      );
+      if (this.uploadFoldersQueue.concurrency !== newConcurrency) {
+        this.uploadFoldersQueue.concurrency = newConcurrency;
+      }
```

Comment:
Good extraction, but maybe we need to refactor this and use a design pattern that fits this case so that the manager does not depend on so many external dependencies and does not give the manager too many responsibilities (not in this task of course)